src/libraries/System.Runtime/tests/System/StringTests.cs

```diff
@@ -1838,5 +1838,39 @@ public void TestStringSearchCacheSynchronization(CompareOptions options)
 
             Task.WaitAll(tasks);
         }
+
+        [Fact]
+        public static void EqualityTests_AsciiOptimizations()
+        {
+            for (int i = 0; i < 128; i++)
+            {
+                for (int j = 0; j < 128; j++)
+                {
+                    bool expectedEqualOrdinal = i == j;
+                    bool expectedEqualOrdinalIgnoreCase = (i == j) || ((i | 0x20) >= 'a' && (i | 0x20) <= 'z' && ((i | 0x20) == (j | 0x20)));
+
+                    string s1 = "aaaaaaa" + (char)i + "bbbbbbb";
```

Comment:
Any particular reason for the prefix/suffix here? Would it be worth also adding a check of comparing the two chars directly (as strings)?

Reply:
The code paths being exercised don't always kick in for small strings. The only way to ensure they get hit is to make the string contents larger than 64 bits.

Reply:
Ah, I see in EqualsIgnoreCase(..) that (in 64 bit) it works in 4 byte chunks, then 2 byte chunks. Your prefixes force both loops. Still, I think it might be worth using varying length prefixes, including 0 -- it's just another loop around this and it will be very fast -- just to ensure we cover all this trickiness?

Reply:
Seems reasonable. In the tests for the SIMD code paths we vary the length like you had proposed. Should be straightforward enough to do the same thing here.